### DISCUSSION TITLE
Use as a Go Module?

### DISCUSSION THREAD

**Discussion by ocomsoft** [prest/prest#684] [API Server] [answered]
I love what you guys have done here. My Go skills are intermediate (I am learning).. I cannot find any documentation anywhere. Is there a way i can use this as a Module? Does anyone have experience doing this?

Maybe the documentation could do with this added in :)

**Comment by avelino** [comment, marked as answer]
We used to encourage developers to extend prestd into a framework format (using prestd's internal packages in their Go project), but this brings some responsibilities to prestd that we are not willing to take on, e.g:
if your project that uses prestd as a package has a security flaw who can guarantee that this flaw isn't caused by prestd, or it could be that the flaw is in your implementation not prestd's and probably prestd will receive an issue that will be almost impossible to simulate without your project so we can fix it

given the above context, we **"remove"** the documentation on how to use prestd in package format and encourage users to use prestd in binary format, so that we can guarantee the project will work

**is it possible to extend prestd?**
yes, prestd has plugin development support where you can develop new endpoints by writing an operating system library (`.so`) - it can be written in Go or any other language that generates a `.so`
[see more here](https://docs.prestd.com/prestd/docs/api-reference/plugins/)

**Comment by ocomsoft** [comment]
Thank you for your detailed response. I saw references to using it as framework but then could not find the pages any more. Now that makes sense :)
